user-management/src/test/java/de/tum/in/www1/artemis/usermanagement/security/jwt/TokenProviderTest.java

```diff
@@ -0,0 +1,104 @@
+package de.tum.in.www1.artemis.usermanagement.security.jwt;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.security.Key;
+import java.util.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import de.tum.in.www1.artemis.security.Role;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import tech.jhipster.config.JHipsterProperties;
+
+class TokenProviderTest {
+
+    private static final long ONE_MINUTE_MS = 60000;
+
+    private Key key;
+
+    private TokenProvider tokenProvider;
+
+    private MockTokenProvider mockTokenProvider;
+
+    @BeforeEach
+    public void setup() {
```

Comment:
Do we really need the mock here?

Reply:
I think yes, because we need a real token that is not expired to validate in  order to test it